### DISCUSSION TITLE
StrongSwan is getting signal 11 on thread 6

### DISCUSSION THREAD

**Discussion by mdmuazzamhusain** [strongswan/strongswan#2096] [General]
Hi Everyone,
Currently i am facing an issue where strongSwan is getting Singnal 11 on thread 6. Killing ourself, received critical signal.
What could be the reason of getting Signal 11.

**Comment by tobiasbrunner** [comment]
> Consider there is no modification in package.

What does that mean?

**Reply by mdmuazzamhusain** [reply]
Hi @tobiasbrunner  ,
Thanks for replying so quicly.
Charon damon is getting killed. below is log and also attached here. 
[log_strongSwan_charon_Kill_ourself.txt](https://github.com/strongswan/strongswan/files/14270893/log_strongSwan_charon_Kill_ourself.txt)

Could you please let me know what are the reson behind this issue. is it a bug in charon or this came by Strongswan server.



2024-02-06T19:39:00.050885+00:00 2620:10d:c085:17:1::143 charon: 06[CHD] updown: up-client-v6:
	2024-02-06T19:39:00.077186+00:00 2620:10d:c085:17:1::143 charon: 06[CHD] updown: ip -6 route add default via fe80::200:5eff:fe00:201 dev br0 proto ra metric 1023 proto static src 2620:10d:c0a8:f4::52
[log_strongSwan_charon_Kill_ourself.txt]
2024-02-06T19:39:00.879604+00:00 2620:10d:c085:17:1::143 charon: 06[CHD] updown: 1 entries updated#015
	**2024-02-06T19:39:00.886485+00:00 2620:10d:c085:17:1::143 charon: 06[DMN] thread 6 received 11**
	2024-02-06T19:39:00.886639+00:00 2620:10d:c085:17:1::143 charon: 06[LIB]  dumping 0 stack frame addresses:
	**2024-02-06T19:39:00.886690+00:00 2620:10d:c085:17:1::143 charon: 06[DMN] killing ourself, received critical signal**

(https://github.com/strongswan/strongswan/files/14270881/log_strongSwan_charon_Kill_ourself.txt)

**Reply by tobiasbrunner** [reply]
No idea. Without backtrace it's difficult to say.

**Reply by mdmuazzamhusain** [reply]
Yes @tobiasbrunner actually backtrace is not generated by charon.
Can you please suggest how can get BT in this case.

below is short log:

charon: 06[DMN] thread 6 received 11
charon: 06[LIB]  dumping 0 stack frame addresses:
charon: 06[DMN] killing ourself, received critical signal

**Reply by tobiasbrunner** [reply]
I guess that depends on the system. There are options for different backtrace libs that can be enabled (search for `backtraces` in [the docs](https://docs.strongswan.org/docs/5.9/install/autoconf.html)). You could also attach GDB to the process if you can reproduce the crash.

**Reply by mdmuazzamhusain** [reply]
I am using 5.3.2 version. It's not reproducible always. GDB is not feasible on on our embedded device due to storage issue.
Can fragmented packets cause this SIGSEGV signal.

**--enable-bfd-backtraces** | 5.0.1 | use binutil’s libbfd to resolve backtraces for memory leaks and segfaults

**Reply by tobiasbrunner** [reply]
That version is very very old, I wouldn't recommend using it. And why would fragmented packages cause this? There was not even a package received (at least according to the log).

**Reply by mdmuazzamhusain** [reply]
@tobiasbrunner   can we use strongSwan 5.9.13 ? is it stable version ?

**Reply by tobiasbrunner** [reply]
Yes, that's the latest stable version.

**Reply by mdmuazzamhusain** [reply]
Hi @tobiasbrunner , currently we are using kernel 3.10.84. So will this 5.9.13 be compatible ?

**Reply by tobiasbrunner** [reply]
Possibly, but no idea.
